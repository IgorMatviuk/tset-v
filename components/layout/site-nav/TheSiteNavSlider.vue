<script>
import { mapState, mapGetters } from 'vuex'

export default {
  data() {
    return {
      offset: 0,
      stack: []
    }
  },
  computed: {
    ...mapState({
      categories: (state) => state.catalog.categories
    }),
    ...mapGetters(['mobileMenu']),
    items() {
      return [
        { name: this.$t('catalogPage.catalog'), categories: this.categories },
        ...this.mobileMenu.columns
      ]
    }
  },
  methods: {
    navigateBack() {
      this.offset -= 1

      this.stack.pop()
    },
    navigateNext(event, item) {
      event.stopPropagation()

      this.offset += 1
      this.stack.push(item)
    },
    getIsPassedItem(item) {
      return this.stack.find((currItem, index) => {
        return currItem.name === item.name && index === this.stack.length - 2
      })
    },
    getIsInactiveItem(item) {
      return !this.stack.includes(item)
    },
    buildRoutePath(item) {
      const stackButRoot = [...this.stack].splice(1)
      const parentPath = stackButRoot
        .filter((stackItem) => stackItem.slug !== item.slug)
        .map((stackItem) => stackItem.slug)

      parentPath.push(item.slug)

      return this.$i18n.path(parentPath.join('/'))
    },
    renderLink(h, item) {
      if (item.categories && item.categories.length) {
        return h(
          'span',
          {
            staticClass: 'nav-slider__link',
            on: {
              click: (event) => this.navigateNext(event, item)
            }
          },
          [
            h('span', item.name),
            h('svg-icon', {
              staticClass: 'nav-slider__icon nav-slider__icon--next',
              props: { name: 'chevron' }
            })
          ]
        )
      }

      return h(
        'NuxtLink',
        {
          props: {
            to: this.buildRoutePath(item),
            activeClass: 'nav-slider__link--active'
          },
          staticClass: 'nav-slider__link'
        },
        item.name
      )
    },
    renderChildList(h, item) {
      return (
        item.categories &&
        h(
          'ul',
          {
            class: {
              'nav-slider--inactive': this.getIsInactiveItem(item),
              'nav-slider--passed': this.getIsPassedItem(item)
            },
            staticClass: 'nav-slider'
          },
          [this.renderItems(h, item.categories)]
        )
      )
    },
    renderBtnBack(h) {
      return h('li', { staticClass: 'nav-slider__item' }, [
        h(
          'span',
          {
            staticClass: 'nav-slider__btn-back',
            on: { click: this.navigateBack }
          },
          [
            h('svg-icon', {
              staticClass: 'nav-slider__icon nav-slider__icon--prev',
              props: { name: 'chevron' }
            }),
            h('span', 'Назад')
          ]
        )
      ])
    },
    renderItems(h, items, isRoot) {
      const renderedItems = items.map((item, index) =>
        h(
          'li',
          {
            staticClass: 'nav-slider__item',
            class: {
              'nav-slider__item--has-children':
                item.categories && item.categories.length
            },
            key: index
          },
          [this.renderLink(h, item), this.renderChildList(h, item)]
        )
      )

      if (isRoot) {
        return renderedItems
      }

      return [this.renderBtnBack(h), ...renderedItems]
    }
  },
  render(h) {
    return h(
      'ul',
      {
        staticClass: 'nav-slider nav-slider--root',
        class: {
          'nav-slider--passed': this.offset !== 0
        },
        ref: 'list'
      },
      [this.renderItems(h, this.items, true)]
    )
  }
}
</script>

<style lang="scss">
.nav-slider {
  @include list-unstyled();

  position: fixed;
  top: 50px;
  right: 0;
  bottom: calc(134px + env(safe-area-inset-bottom));
  left: 0;
  width: 200%;
  padding-right: calc(100% + 25px);
  padding-left: 25px;
  margin-bottom: 0;
  overflow: visible;
  overflow-y: auto;
  background-color: $white;
  transition: left 0.3s cubic-bezier(0.47, 0, 0.745, 0.715);
  -webkit-overflow-scrolling: touch;

  &--passed {
    left: -100%;
  }

  &--inactive {
    left: 100%;
  }

  &__item {
    height: 50px;
    font-size: 12px;
    line-height: 50px;
    text-transform: uppercase;

    &:not(:last-child) {
      line-height: 49px;
      border-bottom: 1px solid #e7e7e7;
    }
  }

  &__link {
    display: flex;
    align-items: center;
    justify-content: space-between;
    letter-spacing: 0.05em;

    &:not(:hover) {
      color: $body-color;
    }

    &:hover {
      text-decoration: none;
    }
  }

  &__btn-back {
    display: flex;
    align-items: center;
    font-weight: 500;
  }

  &__icon {
    width: 10px;
    height: 10px;

    &--prev {
      margin-right: 4px;
      margin-bottom: 2px;
      transform: rotate(90deg);
    }

    &--next {
      transform: rotate(-90deg);
    }
  }

  @include media-breakpoint-up(md) {
    top: 60px;
  }
}
</style>
