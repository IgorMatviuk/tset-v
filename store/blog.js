export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE'
}

export const actionsTypes = {
  GET_BLOG_POSTS: 'GET_BLOG_POSTS'
}

export const state = () => ({
  posts: [],
  isLoading: false,
  currentPage: null,
  totalPages: null
})

export const mutations = {
  [mutationTypes.SET_LOADING_START](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_LOADING_SUCCESS](
    state,
    { posts, currentPage, totalPages }
  ) {
    state.posts = posts
    state.currentPage = currentPage
    state.totalPages = totalPages

    if (state.isLoading) {
      state.isLoading = false
    }
  }
}

export const actions = {
  async [actionsTypes.GET_BLOG_POSTS]({ commit }, params) {
    commit(mutationTypes.SET_LOADING_START, true)

    let query = Object.keys(params)
      .map((key) => `${key}=${encodeURI(params[key])}`)
      .join('&')

    if (!params.limit) {
      query += '&limit=8'
    }

    const { data, meta } = await this.$axios.$get(`api/posts?${query}`)

    const totalPages = Math.ceil(meta.total / (params.limit || 8))

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      posts: data,
      currentPage: meta.current_page,
      totalPages
    })
  }
}
