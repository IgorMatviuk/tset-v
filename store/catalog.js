import { CATALOG_ITEMS_PER_PAGE } from '@/utils/constants'

function createRootCategories(categories) {
  const withRootList = categories.map((category) => {
    if (category.name_alias) {
      const rootCategory = {
        id: category.id,
        name: category.name_alias,
        slug: category.slug
      }

      category.categories = [
        rootCategory,
        ...createRootCategories(category.categories)
      ]
    }

    return category
  })

  return withRootList
}

export const mutationTypes = {
  SET_CATEGORIES: 'SET_CATEGORIES',
  SET_ACTIVE_CATEGORY: 'SET_ACTIVE_CATEGORY',
  SET_PRODUCTS: 'SET_PRODUCTS',
  SET_ATTRIBUTES: 'SET_ATTRIBUTES',
  SET_FILTERS: 'SET_FILTERS',
  CLEAR_FILTERS: 'CLEAR_FILTERS',
  SET_CATALOG_DATA: 'SET_CATALOG_DATA',
  SET_ORDER: 'SET_ORDER',
  SET_SHOW_ITEMS_COUNT: 'SET_SHOW_ITEMS_COUNT',
  SET_LOADING_TOGGLE: 'SET_LOADING_TOGGLE',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE',
  SET_TOTAL_PAGES: 'SET_TOTAL_PAGES',
  RESET_META_DATA: 'RESET_META_DATA',
  SET_SEO_TEXT: 'SET_SEO_TEXT'
}

export const actionTypes = {
  GET_CATEGORIES: 'GET_CATEGORIES',
  GET_PRODUCTS: 'GET_PRODUCTS',
  GET_ATTRIBUTES: 'GET_ATTRIBUTES'
}

export const state = () => ({
  categories: [],
  activeCategory: null,
  products: [],
  attributes: [],
  filters: [],
  order: 'popularity',
  showItemsCount: 4,
  isLoading: false,
  currentPage: null,
  totalPages: null,
  seoText: ''
})

export const mutations = {
  [mutationTypes.SET_CATEGORIES](state, categories) {
    state.categories = categories
  },
  [mutationTypes.SET_ACTIVE_CATEGORY](state, activeCategory) {
    state.activeCategory = activeCategory
  },
  [mutationTypes.SET_PRODUCTS](state, products) {
    state.products = products
  },
  [mutationTypes.SET_ATTRIBUTES](state, attributes) {
    state.attributes = attributes
  },
  [mutationTypes.SET_CATALOG_DATA](state, { category, products, attributes }) {
    state.activeCategory = category
    state.products = products
    state.attributes = attributes
  },
  [mutationTypes.SET_ORDER](state, order) {
    state.order = order
  },
  [mutationTypes.SET_SHOW_ITEMS_COUNT](state, showItemsCount) {
    state.showItemsCount = showItemsCount
  },
  [mutationTypes.SET_FILTERS](state, filters) {
    state.filters = filters
  },
  [mutationTypes.CLEAR_FILTERS](state) {
    state.order = 'popularity'
    state.filters = []
  },
  [mutationTypes.SET_LOADING_TOGGLE](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_TOTAL_PAGES](state, totalPages) {
    state.totalPages = totalPages
  },
  [mutationTypes.RESET_META_DATA](state) {
    state.filters = []
    state.order = 'popularity'
    state.currentPage = 1
  },
  [mutationTypes.SET_SEO_TEXT](state, seoText) {
    state.seoText = seoText
  }
}

export const actions = {
  async [actionTypes.GET_CATEGORIES]({ commit }) {
    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: categories } = await this.$axios.$get(
      `${STAGE_API_URL}/api/categories`
    )

    // const { data: categories } = await this.$axios.$get('/api/categories')
    const normilizedCategories = createRootCategories(categories)

    commit(mutationTypes.SET_CATEGORIES, normilizedCategories)
  },
  async [actionTypes.GET_PRODUCTS]({ commit }, { url, query }) {
    commit(mutationTypes.SET_LOADING_TOGGLE, true)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const {
      data: { products },
      meta
    } = await this.$axios.$get(
      `${STAGE_API_URL}/api/router?url=${url}${query ? `&${query}` : ''}`
    )

    const totalPages = Math.ceil(meta.total / CATALOG_ITEMS_PER_PAGE)

    commit(mutationTypes.SET_PRODUCTS, products)
    commit(mutationTypes.SET_CURRENT_PAGE, meta.current_page)
    commit(mutationTypes.SET_TOTAL_PAGES, totalPages)
    commit(mutationTypes.SET_LOADING_TOGGLE, false)
  }
}

export const getters = {
  defaultCategorySlug(state) {
    // FIXME: temporary solution
    return state.categories[0].slug
  },
  optionLabel(state) {
    const { attributes } = state

    return (code, slug) => {
      const { options } = attributes.find((attr) => attr.code === code)
      const { label } = options.find((option) => option.slug === slug)
      return label
    }
  },
  selectedAttributeOptions(state) {
    const { filters } = state

    return (code) => {
      const filter = filters.find((filter) => filter.code === code)
      return filter ? filter.value : null
    }
  }
}
