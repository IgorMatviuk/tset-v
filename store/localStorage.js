export const mutationTypes = {
  SET_CART_ID: 'SET_CART_ID'
}

export const state = () => ({
  cartId: null,
  status: false
})

export const mutations = {
  [mutationTypes.SET_CART_ID](state, cartId) {
    state.cartId = cartId
  }
}
