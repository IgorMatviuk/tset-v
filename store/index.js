import { actionTypes as catalogActionTypes } from '@/store/catalog'

export const mutationTypes = {
  SET_LANG: 'SET_LANG',
  SET_MENUS: 'SET_MENUS',
  SET_PAGE_TYPE: 'SET_PAGE_TYPE',
  SET_BREADCRUMB: 'SET_BREADCRUMB',
  SET_CONTACTS: 'SET_CONTACTS',
  SET_GLOBALS: 'SET_GLOBALS'
}

export const actionTypes = {
  GET_MENUS: 'GET_MENUS',
  GET_CONTACTS: 'GET_CONTACTS',
  GET_BANNER: 'GET_BANNER',
  GET_CLIENT_LOCATION: 'GET_CLIENT_LOCATION'
}

export const state = () => {
  return {
    locale: 'ru',
    locales: ['uk', 'ru'],
    fallbackLocale: 'ru',
    pageType: null,
    breadcrumb: [],
    menus: [],
    globals: {
      countryCode: 'UA',
      banner: null
    },
    contacts: {
      phone: '0 800 332 512',
      email: 'office@virna.com.ua',
      schedule: 'с 9:00 - 18:00',
      socialLinks: [
        { name: 'instagram', url: 'https://www.instagram.com/' },
        { name: 'facebook', url: 'https://facebook.com/' },
        { name: 'youtube', url: 'https://www.youtube.com/' }
      ]
    }
  }
}

export const mutations = {
  [mutationTypes.SET_LANG](state, locale) {
    state.locale = locale
  },
  [mutationTypes.SET_MENUS](state, menus) {
    state.menus = menus
  },
  [mutationTypes.SET_PAGE_TYPE](state, pageType) {
    state.pageType = pageType
  },
  [mutationTypes.SET_BREADCRUMB](state, breadcrumb) {
    state.breadcrumb = breadcrumb
  },
  [mutationTypes.SET_CONTACTS](state, contacts) {
    state.contacts = contacts
  },
  [mutationTypes.SET_GLOBALS](state, { key, value }) {
    state.globals[key] = value
  }
}

export const actions = {
  async [actionTypes.GET_MENUS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: menus } = await this.$axios.$get(`${STAGE_API_URL}/api/menus`)
    commit(mutationTypes.SET_MENUS, menus)
  },
  async [actionTypes.GET_CONTACTS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: contacts } = await this.$axios.$get(
      `${STAGE_API_URL}/api/contacts`
    )
    commit(mutationTypes.SET_CONTACTS, contacts)
  },
  async [actionTypes.GET_BANNER]({ commit }) {
    const { data: banner } = await this.$axios.$get('/api/banner')
    const defaultBanner = {
      key: 'banner',
      value: banner
    }

    commit(mutationTypes.SET_GLOBALS, defaultBanner)
  },
  async [actionTypes.GET_CLIENT_LOCATION]({ commit }) {
    const locationString = await this.$axios.$get('https://ip2c.org/s')
    const [, countryCode] = locationString.split(';')
    const defaultCountry = {
      key: 'countryCode',
      value: countryCode
    }

    commit(mutationTypes.SET_GLOBALS, defaultCountry)
  },
  async nuxtServerInit({ dispatch }) {
    await Promise.all([
      dispatch(actionTypes.GET_MENUS),
      dispatch(actionTypes.GET_CONTACTS),
      dispatch(`catalog/${catalogActionTypes.GET_CATEGORIES}`)
    ])
  }
}

/**
 * Get menum by type
 * @param {object} state
 * @param {('header'|'footer'|'catalog'|'promo')} type
 */
function getMenuByType(state, type) {
  const { menus } = state
  return menus.find((menu) => menu.menu_type === type)
}

export const getters = {
  mobileMenu(state) {
    return getMenuByType(state, 'mobile')
  },
  desktopMenu(state) {
    return getMenuByType(state, 'header')
  },
  promoMenu(state) {
    return getMenuByType(state, 'promo')
  },
  topMenu(state) {
    return getMenuByType(state, 'top')
  },
  catalogMenu(state) {
    return getMenuByType(state, 'catalog')
  },
  footerMenu(state) {
    return getMenuByType(state, 'footer')
  }
}
