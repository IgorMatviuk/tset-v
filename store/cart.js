import { mutationTypes as localStorageMutationTypes } from '@/store/localStorage'

export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CART_DATA: 'SET_CART_DATA',
  SET_COUNTRIES: 'SET_COUNTRIES',
  SET_REGIONS: 'SET_REGIONS',
  SET_STATES: 'SET_STATES',
  SET_ORDER_SAVE: 'SET_ORDER_SAVE',
  CLEAR_CART: 'CLEAR_CART'
}

export const actionTypes = {
  GET_CHECKOUT_DATA: 'GET_CHECKOUT_DATA',
  GET_CART: 'GET_CART',
  GET_COUNTRIES: 'GET_COUNTRIES',
  GET_REGIONS: 'GET_REGIONS',
  GET_STATES: 'GET_STATES',
  GET_OFFICES: 'GET_OFFICES',
  ADD_ITEM: 'ADD_ITEM',
  UPDATE_ITEM: 'UPDATE_ITEM',
  REMOVE_ITEM: 'REMOVE_ITEM'
}

export const state = () => ({
  cartData: {},
  countries: [],
  regions: [],
  states: [],
  offices: [],
  isOrderSaved: false,
  isLoading: false
})

export const mutations = {
  [mutationTypes.SET_CART_DATA](state, cartData) {
    state.cartData = cartData
  },
  [mutationTypes.SET_LOADING_START](state) {
    state.isLoading = true
  },
  [mutationTypes.SET_LOADING_SUCCESS](state, { key, value }) {
    if (state.isLoading) {
      state.isLoading = false
    }

    state[key] = value
  },
  [mutationTypes.SET_COUNTRIES](state, countries) {
    state.countries = countries
  },
  [mutationTypes.SET_REGIONS](state, regions) {
    state.regions = regions
  },
  [mutationTypes.SET_STATES](state, states) {
    state.states = states
  },
  [mutationTypes.SET_OFFICES](state, offices) {
    state.offices = offices
  },
  [mutationTypes.SET_ORDER_SAVE](state) {
    state.isOrderSaved = true
    state.cartData = {}
  },
  [mutationTypes.CLEAR_CART](state) {
    state.cartData = {}
    state.isOrderSaved = false
  }
}

export const actions = {
  async [actionTypes.GET_CHECKOUT_DATA]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: checkoutData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout`
    )

    commit(mutationTypes.SET_COUNTRIES, checkoutData.countries)
    commit(mutationTypes.SET_REGIONS, checkoutData.regions)
  },
  async [actionTypes.GET_CART]({ commit }, cartId) {
    // const { data: cartData } = await this.$axios.$get('/api/checkout/cart')

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    // const { data: cartData } = await this.$axios.$get(
    //   `${STAGE_API_URL}/api/checkout/cart`
    // )

    const { data: cartData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}`
    )

    if (cartData) {
      commit(mutationTypes.SET_CART_DATA, cartData)
    }
  },
  async [actionTypes.GET_REGIONS]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: regions } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/regions`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'regions',
      value: regions
    })
  },
  async [actionTypes.GET_STATES]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: states } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/states?region_id=${regionId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'states',
      value: states
    })
  },
  async [actionTypes.GET_OFFICES]({ commit }, stateId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: offices } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/offices?state_id=${stateId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'offices',
      value: offices
    })
  },
  async [actionTypes.ADD_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    // const { data: cartData } = await this.$axios.$post(
    //   `/api/checkout/cart/add/${id}`
    // )

    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    const { data: cartData } = await this.$axios.$post(
      `${STAGE_API_URL}/api/checkout/cart/add`,
      {
        cart: cartId,
        variant: itemId
      }
    )

    if (!state.cartData) {
      commit(
        `localStorage/${localStorageMutationTypes.SET_CART_ID}`,
        cartData.id,
        { root: true }
      )
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.UPDATE_ITEM]({ commit }, { cartId, itemId, body }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$put(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/update/${itemId}`,
      body
    )

    // const { data: cartData } = await this.$axios.$put(
    //   `/api/checkout/cart/update/${id}`,
    //   body
    // )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.REMOVE_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$delete(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/remove-item/${itemId}`
    )

    if (!cartData) {
      commit(`localStorage/${localStorageMutationTypes.SET_CART_ID}`, null, {
        root: true
      })
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData || {}
    })
  }
}
