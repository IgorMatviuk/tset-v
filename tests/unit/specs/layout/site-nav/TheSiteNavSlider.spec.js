import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import _cloneDeep from 'lodash.clonedeep'

import TheSiteNavSlider from '@/components/layout/site-nav/TheSiteNavSlider'

import { getters as rootGetters } from '@/store'

import menusFixture from '@/api/fixtures/menus'
import categoriesFixture from '@/api/fixtures/categories'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TheSiteNavSlider', () => {
  let state
  let getters
  let store

  beforeEach(() => {
    state = {
      menus: _cloneDeep(menusFixture),
      catalog: {
        categories: _cloneDeep(categoriesFixture)
      }
    }

    getters = {
      ...rootGetters
    }

    store = new Vuex.Store({
      state,
      getters
    })
  })

  function createSlider() {
    const wrapper = shallowMount(TheSiteNavSlider, {
      localVue,
      store,
      mocks: {
        $t: (str) => str,
        $i18n: {
          path: (str) => str
        }
      },
      stubs: ['svg-icon', 'NuxtLink']
    })

    return wrapper
  }

  test('renders root items', () => {
    const wrapper = createSlider()
    const rootList = wrapper.find('.nav-slider--root')

    expect(wrapper.element).toMatchSnapshot()
    expect(rootList).not.toBeNull()
    expect(rootList.element.tagName).toBe('UL')
    expect(wrapper.findAll('li').length).toBeGreaterThan(0)
  })

  test('renders an items tree', () => {
    const wrapper = createSlider()
    const nestedList = wrapper.find('.nav-slider--root .nav-slider')

    expect(wrapper.vm.categories).not.toBeUndefined()
    expect(nestedList).not.toBeNull()
    expect(nestedList.find('.nav-slider').element).toBeTruthy()
    expect(nestedList.findAll('.nav-slider__item').length).toBeGreaterThan(0)
  })

  test('navigates by items tree', async () => {
    const wrapper = createSlider()
    let itemList = wrapper.find('.nav-slider--root')
    let btnForward = itemList.find('.nav-slider__link')

    expect(itemList.element).toHaveClass('nav-slider')
    expect(btnForward).not.toBeNull()
    expect(btnForward.element.tagName).toBe('SPAN')

    btnForward.trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.offset).toBe(1)

    itemList = wrapper.find(
      '.nav-slider:not(.nav-slider--passed):not(.nav-slider--inactive)'
    )
    btnForward = itemList.find('.nav-slider__link')
    btnForward.trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.offset).toBe(2)

    const btnBack = wrapper.find('.nav-slider__btn-back')
    btnBack.trigger('click')
    expect(wrapper.vm.offset).toBe(1)
  })
})
