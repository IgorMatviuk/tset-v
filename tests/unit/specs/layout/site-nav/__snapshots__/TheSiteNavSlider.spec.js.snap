// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSiteNavSlider renders root items 1`] = `
<ul
  class="nav-slider nav-slider--root"
>
  <li
    class="nav-slider__item nav-slider__item--has-children"
  >
    <span
      class="nav-slider__link"
    >
      <span>
        catalogPage.catalog
      </span>
      <svg-icon-stub
        class="nav-slider__icon nav-slider__icon--next"
      />
    </span>
    <ul
      class="nav-slider nav-slider--inactive"
    >
      <li
        class="nav-slider__item"
      >
        <span
          class="nav-slider__btn-back"
        >
          <svg-icon-stub
            class="nav-slider__icon nav-slider__icon--prev"
          />
          <span>
            Назад
          </span>
        </span>
      </li>
      <li
        class="nav-slider__item nav-slider__item--has-children"
      >
        <span
          class="nav-slider__link"
        >
          <span>
            Платья
          </span>
          <svg-icon-stub
            class="nav-slider__icon nav-slider__icon--next"
          />
        </span>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Повседневные платья
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Нарядные платья
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Летние платья
            </nuxtlink-stub>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Сарафаны
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item nav-slider__item--has-children"
      >
        <span
          class="nav-slider__link"
        >
          <span>
            Костюмы
          </span>
          <svg-icon-stub
            class="nav-slider__icon nav-slider__icon--next"
          />
        </span>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Повседневные костюмы
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Классические костюмы
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Спортивные костюмы
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Летние костюмы
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Комбинезоны
            </nuxtlink-stub>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Брюки и шорты
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Юбки
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Гольфы
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Блузы и рубашки
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item"
      >
        <nuxtlink-stub
          class="nav-slider__link"
        >
          Свитеры и джемперы
        </nuxtlink-stub>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="nav-slider__item nav-slider__item--has-children"
      >
        <span
          class="nav-slider__link"
        >
          <span>
            Верхняя одежда
          </span>
          <svg-icon-stub
            class="nav-slider__icon nav-slider__icon--next"
          />
        </span>
        <ul
          class="nav-slider nav-slider--inactive"
        >
          <li
            class="nav-slider__item"
          >
            <span
              class="nav-slider__btn-back"
            >
              <svg-icon-stub
                class="nav-slider__icon nav-slider__icon--prev"
              />
              <span>
                Назад
              </span>
            </span>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Пальто
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Пиджаки
            </nuxtlink-stub>
          </li>
          <li
            class="nav-slider__item"
          >
            <nuxtlink-stub
              class="nav-slider__link"
            >
              Тренчи
            </nuxtlink-stub>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    class="nav-slider__item"
  >
    <nuxtlink-stub
      class="nav-slider__link"
    >
      О нас
    </nuxtlink-stub>
  </li>
  <li
    class="nav-slider__item"
  >
    <nuxtlink-stub
      class="nav-slider__link"
    >
      Sale
    </nuxtlink-stub>
  </li>
  <li
    class="nav-slider__item"
  >
    <nuxtlink-stub
      class="nav-slider__link"
    >
      АКЦИИ
    </nuxtlink-stub>
  </li>
</ul>
`;
